src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java

```diff
@@ -67,6 +68,23 @@ void should_not_process_file_if_it_is_not_ready_yet() {
         verify(blobDispatcher, never()).dispatch(any(), any(), any(), any());
     }
 
+    @Test
+    void should_not_store_envelope_in_db_when_upload_failed() {
+        // given
+        blobExists(OffsetDateTime.now());
+        given(readinessChecker.isReady(any())).willReturn(true);
+
+        willThrow(new RuntimeException("Test exception"))
+            .given(blobDispatcher)
+            .dispatch(any(), any(), any(), any());
+
+        // when
+        blobProcessor.process("envelope.zip", "container1");
+
+        // then
+        verify(envelopeRepo, never()).insert(any());
```

Comment:
may be worth verifying that method throwing exception was called

Reply:
Good idea, done